test/UnitTests/CheckoutControllerTest.cs

```diff
@@ -0,0 +1,168 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Security.Claims;
+using System.Threading;
+using Microsoft.AspNet.Http.Core;
+using Microsoft.AspNet.Http.Core.Collections;
+using Microsoft.AspNet.Mvc;
+using Microsoft.AspNet.Routing;
+using Microsoft.Framework.DependencyInjection;
+using Microsoft.Framework.DependencyInjection.Fallback;
+using Xunit;
+using MusicStore.Controllers;
+using MusicStore.Models;
+
+namespace UnitTests
+{
+    public class CheckoutControllerTest
+    {
+        IServiceProvider _serviceProvider;
+
+        public CheckoutControllerTest()
+        {
+            var services = new ServiceCollection();
+
+            services.AddEntityFramework()
+                      .AddInMemoryStore()
+                      .AddDbContext<MusicStoreContext>();
+
+            _serviceProvider = services.BuildServiceProvider();
+        }
+
+        [Fact]
+        public void AddressAndPaymente_ReturnsDefaultView()
+        {
+            // Arrange
+            var controller = new CheckoutController();
+
+            // Act
+            var result = controller.AddressAndPayment();
+
+            // Assert
+            var viewResult = Assert.IsType<ViewResult>(result);
+            Assert.Null(viewResult.ViewName);
+        }
+
+        [Fact]
+        public void AddressAndPayment_ReturnsOrderIfInvalidPromoCode()
+        {
+            // Arrange
+            var context = new DefaultHttpContext();
+            context.Request.Form =
+                new FormCollection(new Dictionary<string, string[]>());
```

Comment:
>  I didn't include that test in this PR. I'll make sure to add that test after figuring out how to handle Session.

Seems fine.

BTW I think the best way forward here is not to mock session, it's to mock the shopping cart. Let's discuss after completing the current round